### PR TITLE
Introduce 2d intersection

### PR DESCRIPTION
**Pull request by asalzburger**
This PR extends the `Intersection` class family to describe both 2D and 3D intersections. It is a preparation of the new digitisation model, that will be done only in the projective plane, rather than in full 3D for simplicity.

The intersection will now follow the naming convention of `VectorXD`, i.e.

```c++
using Intersection2D = Intersection<2>;
using Intersection3D= Intersection<3>;
```

It also introduced a `IntersectionHelper2D`, which will be used in the digitisation (and potentially union/intersection building), together with a set of UnitTests that the intersection maths is correct.

### CODE REVIEW

**Comment by asalzburger** [comment]
Hi @Corentin-Allaire - this is a rather small PR, could you please review?

File: Core/include/Acts/Surfaces/detail/IntersectionHelper2D.hpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+namespace Acts {
+namespace detail {
+
+struct IntersectionHelper2D {
+  /// Intersect two segments
+  ///
+  /// @param s0 The Start of the segement
+  /// @param s1 The end of the segement
+  /// @param origin The Start of intersection line
+  /// @param direction The Direction of intersection line
```
Comment (Corentin-Allaire) [inline Core/include/Acts/Surfaces/detail/IntersectionHelper2D.hpp:25]: Maybe you should add a return line similar to the one for ellipses and circle ?

File: Tests/UnitTests/Core/Surfaces/IntersectionHelper2DTests.cpp [hunk 2]
```diff
@@ -0,0 +1,185 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2018 CERN for the benefit of the Acts project
```
Comment (Corentin-Allaire) [inline Tests/UnitTests/Core/Surfaces/IntersectionHelper2DTests.cpp:3]: The Copyright date should be 2020

File: Core/include/Acts/Surfaces/detail/IntersectionHelper2D.hpp (hunk 1 above)
Comment (asalzburger) [reply Core/include/Acts/Surfaces/detail/IntersectionHelper2D.hpp:25]: Fixed.

File: Tests/UnitTests/Core/Surfaces/IntersectionHelper2DTests.cpp (hunk 2 above)
Comment (asalzburger) [reply Tests/UnitTests/Core/Surfaces/IntersectionHelper2DTests.cpp]: Alright.

**Pull request merged**